_posts/2019-11-14-diagnosing-our-radiation-problems.markdown

```diff
@@ -0,0 +1,118 @@
+---
+layout: epic
+title: "Diagnosing our Radiation Problems"
+date: "2019-11-14"
+author: [ash]
+categories: [databases, encryption, email, team, people, best practices, culture]
+---
+
+Email! Electronic mail! What a concept! Like many companies, Artsy has built products on top of email, but this is
+a decision that (like many companies) Artsy periodically regrets. But overall, our email systems work well!
+
+But what about when it doesn't? Well that's what today's blog post is about: what happens when things break and you
+don't know why?
+
+<!-- more -->
+
+I have learned a lot
+[since my first on-call shift](https://artsy.github.io/blog/2018/05/30/my-first-week-on-call/), but going on call
+still gives me a little stage fright. As I start a shift, I'm on the look-out for things that might break, and soon
+after starting a recent on-call shift, "breaking" is exactly what things did.
+
+![Screenshot of Slack alert](/images/2019-11-07-diagnosing-our-radiation-problems/alert.png)
+
+We got an automated alert on Slack that one of our email services, code-named "Radiation", had really high response
+times. More specifically, it was timing out. Yikes. The next twenty two hours was a deep dive into Heroku, New
```

Comment:
I think we might want to rephrase this a bit to reflect a few different alerting/error conditions.

1. We received the "Response times (web)" alert from Heroku. This alert told us that for completed requests, an elevated amount were completing longer than our configured threshold.
2. We also received the "Failed requests (web)" alert from Heroku. This alert told us that a percentage of all requests were failing, above a configured threshold.
3. Then we saw that some clients were hitting request timeouts, particularly from the Heroku router.

It think it's an important distinction that Radiation wasn't timing out requests or timing out, but rather that the client (the Heroku router or a downstream client) would give up after a configured "timeout" threshold.